File: src/main/java/com/market/wanted/order/entity/Order.java
package com.market.wanted.order.entity;

import com.market.wanted.common.entity.BaseEntity;
import com.market.wanted.member.entity.Member;
import com.market.wanted.product.entity.ProductStatus;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import static jakarta.persistence.FetchType.LAZY;

@Getter
@Entity
@Table(name = "orders")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Order extends BaseEntity {
    @Id
    @Column(name = "order_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private OrderStatus orderStatus;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "seller_id")
    private Member seller;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "buyer_id")
    private Member buyer;

    @OneToOne(mappedBy = "order", cascade = CascadeType.ALL)
    private OrderItem orderItem;

    public Order(OrderStatus orderStatus, Member buyer, Member seller) {
        this.orderStatus = orderStatus;
        this.seller = seller;
        this.buyer = buyer;
    }

    public void addOrderItem(OrderItem orderItem) {
        this.orderItem = orderItem;
        orderItem.setOrder(this);

    }

    public void confirm() {
        this.orderStatus = OrderStatus.COMPLETE;
        orderItem.getProduct().changeStatus(ProductStatus.SOLD_OUT);
    }

    public void addBuyer(Member buyer) {
        this.buyer = buyer;
    }
}

File: src/main/kotlin/wanted/market/api/order/domain/entity/Order.kt
package wanted.market.api.order.domain.entity

import jakarta.persistence.*
import jakarta.persistence.EnumType.STRING
import jakarta.persistence.FetchType.LAZY
import jakarta.persistence.GenerationType.IDENTITY
import org.springframework.boot.context.properties.bind.DefaultValue
import wanted.market.api.common.BaseTimeEntity
import wanted.market.api.member.domain.entity.Member
import wanted.market.api.order.domain.entity.OrderStatus.*

@Entity
class Order(

    @OneToMany(mappedBy = "orders")
    val orderItems: MutableList<OrderItem> = mutableListOf(),

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "buyer_id")
    val buyer: Member,

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "seller_id")
    val seller: Member,

    @Enumerated(STRING)
    @DefaultValue("NONE")
    var orderStatus: OrderStatus = NONE,

    @Id
    @GeneratedValue(strategy = IDENTITY)
    @Column(name = "order_id")
    val id: Long? = null
) : BaseTimeEntity<Order, Long>() {

    fun approveOrder() {
        this.orderStatus = SALEAPPROVAL
    }

    fun confirmOrder() {
        this.orderStatus = PURCHASECONFIRM
    }

    companion object {
        fun create(buyer: Member, seller: Member): Order {
            return Order(
                buyer = buyer,
                seller = seller
            )
        }
    }
}